packages/react-core/rollup.config.js

```diff
@@ -1,36 +1,42 @@
-// rollup.config.js
 import { defineConfig } from 'rollup';
 import typescript from '@rollup/plugin-typescript';
 import externals from 'rollup-plugin-node-externals';
 
+// common config settings
+const input = ['src/index.ts'];
+const sourceMap = false;
+const tsconfig = 'tsconfig.dist.json';
+
+const esmOutputDir = 'dist/esm';
+
 const config = defineConfig([
   // CJS config
   {
-    input: ['src/index.ts'],
-    output: {
-      dir: 'dist',
-      format: 'cjs',
-      sourcemap: false,
-    },
+    input,
+    output: { dir: 'dist', format: 'cjs' },
     plugins: [
       externals({ include: /^@aws-amplify/ }),
-      typescript({ declarationDir: 'dist/types', sourceMap: false }),
+      typescript({ declarationDir: 'dist/types', sourceMap, tsconfig }),
     ],
   },
   // ESM config
   {
-    input: ['src/index.ts'],
+    input,
     output: {
-      dir: 'dist/esm',
+      dir: esmOutputDir,
       format: 'es',
       entryFileNames: '[name].mjs',
       preserveModules: true,
       preserveModulesRoot: 'src',
-      sourcemap: false,
```

Comment:
Removed the `output.sourcemap` option in all of the ESM and CJS rollup configs as they are extraneous. The setting of `sourceMap` to `false` in the `typescript` plugin options handles it